### PR TITLE
Add Resource Process Package

### PR DESCRIPTION
**Pull request by gz65555**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)

Add `@oasis-engine/resource-process` package for editor file process.

### CODE REVIEW

**Review by GuoLei1990 — APPROVED** [review]
+1

File: packages/resource-process/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  });
+}
+
+/**
+ * Encode main file and meta to an array buffer.
+ * @param main - main file data
+ * @param meta - meta file data
+ * @param byteLength - byte length of estimate array buffer
+ */
+export function encode(main: any, meta: any, byteLength?: number): ArrayBuffer {
```
Comment (zhuxudong) [inline packages/resource-process/src/index.ts:35]: `byteLength` can't be ignored

File: packages/resource-process/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
```
Comment (zhuxudong) [inline packages/resource-process/src/index.ts:35]: are `meta` have no type ?

**Review by zhuxudong — APPROVED** [review]
+1

**Pull request merged**